static/src/javascripts/projects/common/modules/support/banner.ts

```diff
@@ -0,0 +1,205 @@
+import { mountDynamic } from '@guardian/automat-modules';
+import { log, storage } from '@guardian/libs';
+import {
+	getBanner,
+	getPuzzlesBanner,
+	getWeeklyArticleHistory,
+} from '@guardian/support-dotcom-components';
+import type {
+	BannerPayload,
+	ModuleDataResponse,
+} from '@guardian/support-dotcom-components/dist/dotcom/src/types';
+import type { BannerTargeting } from '@guardian/support-dotcom-components/dist/shared/src/types/targeting';
+import type React from 'react';
+import { trackNonClickInteraction } from 'common/modules/analytics/google';
+import { getMvtValue } from 'common/modules/analytics/mvt-cookie';
+import { submitComponentEvent } from 'common/modules/commercial/acquisitions-ophan';
+import { getVisitCount } from 'common/modules/commercial/contributions-utilities';
+import { shouldHideSupportMessaging } from 'common/modules/commercial/user-features';
+import {
+	buildTagIds,
+	dynamicImport,
+	getArticleCountConsent,
+	isHosted,
+	ModulesVersion,
+	supportDotcomComponentsUrl,
+	tracking,
+} from 'common/modules/support/supportMessaging';
+import userPrefs from 'common/modules/user-prefs';
+import config from 'lib/config';
+import fastdom from 'lib/fastdom-promise';
+import { getCountryCode } from 'lib/geolocation';
+import reportError from 'lib/report-error';
+
+export const NO_RR_BANNER_TIMESTAMP_KEY = 'gu.noRRBannerTimestamp'; // timestamp of when we were last told not to show a RR banner
+const twentyMins = 20 * 60_000;
+
+export const withinLocalNoBannerCachePeriod = (): boolean => {
+	const item = window.localStorage.getItem(NO_RR_BANNER_TIMESTAMP_KEY);
+	if (item && !Number.isNaN(parseInt(item, 10))) {
+		const withinCachePeriod = parseInt(item, 10) + twentyMins > Date.now();
+		if (!withinCachePeriod) {
+			// Expired
+			window.localStorage.removeItem(NO_RR_BANNER_TIMESTAMP_KEY);
+		}
+		return withinCachePeriod;
+	}
+	return false;
+};
+
+export const setLocalNoBannerCachePeriod = (): void =>
+	window.localStorage.setItem(NO_RR_BANNER_TIMESTAMP_KEY, `${Date.now()}`);
+
+export const renderBanner = (
+	response: ModuleDataResponse,
+): Promise<boolean> => {
+	if (!response.data) {
+		return Promise.resolve(false);
+	}
+	const { module, meta } = response.data;
+
+	return dynamicImport(module.url, module.name)
+		.then((Banner: React.FC<any>) => {
```

Comment:
To avoid an `any`, you can just let the default `{}`.

```suggestion
		.then((Banner: React.FC) => {
```